parallel-consumer-core/src/main/java/io/confluent/csid/utils/Range.java

```diff
@@ -57,23 +76,17 @@ public void remove() {
         };
     }
 
-    public List<Integer> list() {
-        ArrayList<Integer> integers = new ArrayList<Integer>();
-        forEach(integers::add);
+    /**
+     * Potentially slow, but useful for tests
+     */
+    public List<Integer> listAsIntegers() {
+        ArrayList<Integer> integers = new ArrayList<>();
+        forEach(e -> integers.add(Math.toIntExact(e)));
```

Comment:
wouldnt that be slow? or its just to avoid cast warning?
can just test 
```
if(limit<Integer.MAX_VALUE){
     forEach(e -> integers.add((int)e));
}
```
or alternatively - integers.addAll(IntStream.range(0,limit).Collectors.toList()); -- if need to ensure ArrayList / i.e. list with Add support etc...
or just IntStream.range(0,limit).Collectors.toList() if dont care if can add to it later...